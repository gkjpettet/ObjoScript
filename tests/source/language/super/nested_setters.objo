class Person {
name=(value) {
_name = value
System.print("From Person instance")
}

print() {
System.print(_name)
}
}

class Doctor is Person {
name=(value) {
super.name = value
System.print("From Doctor instance")
}
}

class Radiologist is Doctor {
name=(value) {
super.name = value
System.print("From Radiologist instance")
}
}

var aoife = Person()
aoife.name = "Aoife"
aoife.print()

var fi = Doctor()
fi.name = "Fiona"
fi.print()

var garry = Radiologist()
garry.name = "Garry"
garry.print()