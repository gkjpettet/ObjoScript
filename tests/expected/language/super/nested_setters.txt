From Person instance
Aoife
From Person instance
From Doctor instance
Fiona
From Person instance
From Doctor instance
From Radiologist instance
Garry